libraries/aws_rds_snapshot.rb

```diff
@@ -17,7 +17,7 @@ def initialize(opts = {})
     super(opts)
     validate_parameters(required: [:db_snapshot_identifier])
 
-    raise ArgumentError, "#{@__resource_name__}: db_snapshot_identifier must start with a letter followed by up to 62 letters/numbers/hyphens." if opts[:db_snapshot_identifier] !~ /^[a-z]{1}[0-9a-z\-]{0,62}$/
+    raise ArgumentError, "#{@__resource_name__}: db_snapshot_identifier must start with a letter followed by up to 1 to 255 characters." if opts[:db_snapshot_identifier] !~ /^[a-z]{1}[0-9a-z\-\:]{1,255}$/
```

Comment:
please review the recommendation from here https://github.com/inspec/inspec-aws/pull/622#issuecomment-917853734

Reply:
this is needed to support system generated snapshots.